cf/net/gateway.go

```diff
@@ -362,33 +362,21 @@ func (gateway Gateway) doRequestHandlingAuth(request *Request) (*http.Response,
 		httpReq.Body = ioutil.NopCloser(request.SeekableBody)
 	}
 
-	// perform request
-	rawResponse, err := gateway.doRequestAndHandlerError(request)
-	if err == nil || gateway.authenticator == nil {
-		return rawResponse, err
-	}
-
-	switch err.(type) {
-	case *errors.InvalidTokenError:
-		// refresh the auth token
-		var newToken string
-		newToken, err = gateway.authenticator.RefreshAuthToken()
-		if err != nil {
-			return rawResponse, err
-		}
-
-		// reset the auth token and request body
-		httpReq.Header.Set("Authorization", newToken)
-		if request.SeekableBody != nil {
-			_, _ = request.SeekableBody.Seek(0, 0)
-			httpReq.Body = ioutil.NopCloser(request.SeekableBody)
+	if gateway.authenticator != nil {
+		// in case the request is purposefully unauthenticated (invocation without prior login)
+		// do not attempt to refresh the token (it does not exist)
+		if request.HTTPReq.Header.Get("Authorization") != "" {
```

Comment:
I implemented the approach you proposed.
I added the corresponding `RefreshToken(token string)` rather than changing `RefreshAuthToken()` since it is used quite heavily all over the project.

Reply:
Looks good, thanks!